### ISSUE TITLE
Why we can't make production releases yet

### ISSUE THREAD

**Issue by AucaCoyan** [federico-terzi/espanso#2126]
### Description

Hi! You are probably reading this because someone gave you a link of this explanation.
The thing is `espanso` was almost entirely developed by @federico-terzi , the original maintainer of this project.
Time passed and he couldn't keep up with issues, new features and other personal projects. That's why he made the call to new team members in #1742 , and many people showed up, including @smeech and me, @AucaCoyan (current maintainers).

Now here lies the problem, _because_ Federico used his personal signing key to release the code (that is, a signature that he _wrote and reviewed_ the code) this meets a conflict for making espanso an organization of developers (The newer people). For many reasons he is unable to spent time in the project, and would be rude to push code to production with his signature in the process, but not having it reviewed by him.

So, in conclusion we are migrating from a personal signing key into an organization signing key, and the process can be followed up in #1928 .

If you have any doubts, [join the discord](https://discord.gg/DFcCNDg7bB)

**Comment by j-lakeman** [comment]
I guess this applies to the hub repo similarly? There are quite some PRs awaiting approval by someone with write access.

**Comment by AucaCoyan** [comment]
Not completely:
- if you are speaking of [`espanso/hub`](https://github.com/espanso/hub), we do have write access, and usually smeech is working to review every single PR submitted.
- if instead you are referring to [`espanso/hub-frontend`](https://github.com/espanso/hub-frontend) I do not have write access, it needs an approval of a maintainer, @arabello in this scenario

**Comment by j-lakeman** [comment]
The former. Awesome!
If I was to contribute I should probably fork [espanso/hub](https://github.com/espanso/hub), right?
E.g. the little GitHub icon [here](https://hub.espanso.org/contractions-en) points to the repo of the initial author rather than the hub repo which confuses me slightly.

**Comment by smeech** [comment]
> If I was to contribute I should probably fork [espanso/hub](https://github.com/espanso/hub), right?

That's right.

The GitHub icon in the package page in the Hub points to the author's homepage, as specified in the `_manifest.yml` in their package.

I have merged a [lot](https://github.com/espanso/hub/pulls?q=is%3Apr+is%3Aclosed) of packages this year. The only ones remaining have issues, major or minor, that need fixing before they're merged, or have outstanding questions.

**Comment by arabello** [comment]
> if instead you are referring to [espanso/hub-frontend](https://github.com/espanso/hub-frontend) I do not have write access, it needs an approval of a maintainer, @arabello in this scenario

I provided write access to @AucaCoyan some time ago, are you sure you don't have permissions? If so, let me know what I can do

**Comment by AucaCoyan** [comment]
Apparently I can't. In [this PR](https://github.com/espanso/hub-frontend/pull/49) requires a review of a maintainer and I think you are the only other one who can review. This should be a setting of this specific repo, as in another repos of the org this doesn't happen
I don't see the settings tab either, so I guess I may have write permissions but unable to merge PRs due to the issue of above

**Comment by smeech** [comment]
Perhaps you can't review that one because it's your own commit?

**Comment by AucaCoyan** [comment]
> Perhaps you can't review that one because it's your own commit?

That's what I thought, I couldn't test this theory with another PR, but it's common to have that policy in enterprise repos (not specifically GitHub, but anywhere)

**Comment by arabello** [comment]
yes, it's due to the protection rule on the branch. You have write permissions, but in any case a PR must be reviewed by someone else with write permissions.

The PR is just a version update so I'll approve it on trust. To address the problem we might want
- to give someone else write permissions
- remove the protection rule (I don't like this)
- hard ping me if some PRs are blocking considering the low rate of PRs creation from member with write permissions

**Comment by AucaCoyan** [comment]
Hey @arabello !
Sorry for the late response, this went down in my to-do list 😁.

> Yes, it's due to the protection rule on the branch. You have write permissions, but in any case a PR must be reviewed by someone else with write permissions.

This is expected in work environments, or at least teams with a couple of developers. I would like to apply that protection for our repos, but the reality tell us that I'm the sole developer of the organization (only smeech has write permissions and he doesn't fully understand all the lines of code). Under a normal work-environment, branch protections should be enforced because you have at least a boss to review your code, but in espanso we don't meet the minimum requirement of "two developers checking the notifications daily" to that protection policy being helpful instead of a barrier.

Having said that, is it too much to ask to remove the branch protection?
I know it's not the perfect solution, but I believe it's for the better for a speedy (and most importantly, not upsetting) development.

**Comment by arabello** [comment]
> Having said that, is it too much to ask to remove the branch protection?

@AucaCoyan Not at all, we could remove the branch protection. However, branch protection (require approvals) avoids accidental merges and merges from people with no context of the project.

If the issue is you being able to autonomously merge PR, I simply add you (already done) as "[Allow specified actors to bypass required pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#:~:text=Optionally%2C%20to%20allow%20specific%20actors%20to%20push%20code%20to%20the%20branch%20without%20creating%20pull%20requests%20when%20they%27re%20required%2C%20select%20Allow%20specified%20actors%20to%20bypass%20required%20pull%20requests.%20Then%2C%20search%20for%20and%20select%20the%20actors%20who%20should%20be%20allowed%20to%20skip%20creating%20a%20pull%20request.)". This allows you to ignore every PR requirement. If this doesn't work I can always make you an admin so to bypass branch protection. I think this is safer for the project, but let me know what do you think and I will surely follow what you need.

Note: also note that as a person with write permission you _should_ be able to push on `main` with no PR (I don't remember if you have to be admin or just write permission)

**Comment by AucaCoyan** [comment]
Since I can sign releases and SignPath is donating its infrastructure to sign Windows executables, this was solved 🎉 !
